src/datachain/lib/listing.py

```diff
@@ -97,15 +109,68 @@ async def _scandir(client, prefix, fetch_workers) -> AsyncIterator:
     await main_task
 
 
-def list_bucket(uri: str, client_config=None, fetch_workers=FETCH_WORKERS) -> Callable:
+def list_bucket(uri: str, fetch_workers=FETCH_WORKERS, **kwargs) -> Callable:
     """
     Function that returns another generator function that yields File objects
     from bucket where each File represents one bucket entry.
     """
 
     def list_func() -> Iterator[File]:
-        config = client_config or {}
-        client, path = Client.parse_url(uri, None, **config)  # type: ignore[arg-type]
+        client, path = Client.parse_url(uri, None, **kwargs)  # type: ignore[arg-type]
         yield from iter_over_async(_scandir(client, path, fetch_workers), get_loop())
 
     return list_func
+
+
+def ls(
+    dc: "DataChain",
+    path: str,
+    recursive: Optional[bool] = True,
+    object_name="file",
+):
+    """
+    Return files by some path from DataChain instance which contains bucket listing.
+    Path can have globs.
+    If recursive is set to False, only first level children will be returned by
+    specified path
+    """
+
+    def _file_c(name: str) -> Column:
+        return Column(f"{object_name}.{name}")
+
+    dc = dc.filter(_file_c("is_latest") == true())
+
+    if recursive:
+        if not path or path == "/":
+            # root of a bucket, returning all latest files from it
+            return dc
+
+        if not glob.has_magic(os.path.basename(os.path.normpath(path))):
+            # path is not glob, so it's pointing to some directory or a specific
+            # file and we are adding proper filter for it
+            return dc.filter(
+                (_file_c("path") == path)
+                | (_file_c("path").glob(path.rstrip("/") + "/*"))
+            )
+
+        # path has glob syntax so we are returning glob filter
+        return dc.filter(_file_c("path").glob(path))
+    # returning only first level children by path
+    return dc.filter(pathfunc.parent(_file_c("path")) == path.lstrip("/").rstrip("/*"))
+
+
+def listing_dataset_name(uri: str, path: str) -> str:
+    """Returns special listing dataset name from uri and path"""
+    return f"{LISTING_PREFIX}{uri}/{path.lstrip('/')}"
```

Comment:
important: does dataset name always end with `/` or not? we need put some logic to always check this rule and / or assert on it (it affects the listing_subset - if we always have `/` at the end `startswith` might be enough, otherwise it should be a different implementation

Reply:
Yes, it should always end up with `/`. This might change if we allow enlisting single file though but that's separate issue anyway and for now we always enlist directory.
I will add some additional checks for this

Reply:
yep, let's put asserts to document it everywhere to enforce this and catch if it's broken